### ISSUE TITLE
PPO2 - Why is Value Function loss divided by 2

### ISSUE THREAD

**Issue by harrismirza** [hill-a/stable-baselines#106]
Looking through the PPO2 code, I noticed that the Value function loss is divided by 2, why is this the case?

**Comment by araffin** [comment]
Hello,
I would answer in 2 points:
first, we just kept the code from OpenAI (but that does not really answer the question)
https://github.com/hill-a/stable-baselines/blob/6c6d36a6a81173ccb2ee2ab343c5cf9dc5f4b832/stable_baselines/ppo2/ppo2.py#L139

See original code: https://github.com/openai/baselines/blob/master/baselines/ppo2/model.py#L72

Then, I would say this is common practice when you use the mean squared error (see https://math.stackexchange.com/questions/884887/why-divide-by-2m), when you derive, the factor 2 goes away.

**Comment by harrismirza** [comment]
I think it would make a difference when optimizing because by the same logic the coefficients for both the value and entropy loss would also make no difference. Maybe I'll ask on the OpenAI Repo.

**Comment by araffin** [comment]
>by the same logic the coefficients for both the value and entropy loss would also make no difference

Can you elaborate more?

**Comment by harrismirza** [comment]
If we think about the loss function that is calculated and passed to the optimizer from the code:

https://github.com/hill-a/stable-baselines/blob/6c6d36a6a81173ccb2ee2ab343c5cf9dc5f4b832/stable_baselines/ppo2/ppo2.py#L134-L147

Let's say you ignore the fact the value loss is divided by 2 when you assign it to the variables then the loss function is basically 
`
loss = pg_loss - (entropy * ent_coef) + (vf_loss * vf_coef * 0.5)
`

Therefore as both 0.5 and vf_coef are constants (well vf_coef can change based on the episode number, but it acts as a constant here), then if the 0.5 doesn't matter then the vf_coef doesn't either.

**Comment by araffin** [comment]
>if the 0.5 doesn't matter then the vf_coef doesn't either.

I see your point but I disagree in a sense that this is just a matter of scaling. That is to say, if we remove the 0.5 factor, then we just need to scale down vf_coeff to obtain the same behavior.

**Comment by harrismirza** [comment]
Ok, thanks for the help.
